tools/tsp-client/ci.yml

```
@@ -0,0 +1,120 @@
+# Node.js
+# Build a general Node.js project with npm.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+
+trigger:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/tsp-client
+
+pr:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/tsp-client
+
+variables:
+  - template: ../../eng/pipelines/templates/variables/globals.yml
+  - name: NodeVersion
+    value: '14.x'
+  - name: VAR_ARTIFACT_NAME
+    value: 'drop'
+  - name: VAR_BUILD_ARTIFACT_STAGING_DIRECTORY
+    value: $(Build.ArtifactStagingDirectory)
+
+stages:
+  - stage: InstallAndBuild
+    jobs:
+      - job: Build
+        strategy:
+          matrix:
+            linux:
+              imageName: ubuntu-22.04
+            mac:
+              imageName: 'macos-latest'
+            windows:
+              imageName: 'windows-latest'
+        pool:
+          vmImage: $(imageName)
```

This will update it to use our custom managed image pools and be more in line with convention.

```suggestion
              OSVmImage: 'MMSUbuntu20.04'
              Pool: 'azsdk-pool-mms-ubuntu-2004-general'
            mac:
              OSVmImage: 'macos-latest'
              Pool: 'Azure Pipelines'
            windows:
              OSVmImage: 'MMS2022'
              Pool: 'azsdk-pool-mms-win-2022-general'
        pool:
          name: $(Pool)
          vmImage: $(OSVmImage)
```